lib/autoupdate/agent/process.go

```diff
@@ -555,3 +559,22 @@ func (c *localExec) Run(ctx context.Context, name string, args ...string) (int,
 	code := cmd.ProcessState.ExitCode()
 	return code, trace.Wrap(err)
 }
+
+// Output runs the command and returns combined output from stdout and stderr.
+// Same arguments as exec.CommandContext.
+func (c *localExec) Output(ctx context.Context, name string, args ...string) ([]byte, error) {
+	var buf bytes.Buffer
+	stderrLogger := &lineLogger{ctx: ctx, log: c.Log, level: c.ErrLevel, prefix: "[stderr] "}
+	stdoutLogger := &lineLogger{ctx: ctx, log: c.Log, level: c.OutLevel, prefix: "[stdout] "}
+
+	cmd := exec.CommandContext(ctx, name, args...)
+	cmd.Dir = c.Dir
+	stderr := io.MultiWriter(&buf, stderrLogger)
+	stdout := io.MultiWriter(&buf, stdoutLogger)
```

Comment:
`byte.Buffer.Write` is not reentrant

Reply:
As discussed on Slack, this is fixed in upstream os/exec so we should be good here.